core/src/main/java/org/apache/gravitino/storage/relational/service/ModelVersionMetaService.java

```diff
@@ -321,26 +329,53 @@ public <E extends Entity & HasIdentifier> ModelVersionEntity updateModelVersion(
     try {
       SessionUtils.doMultipleWithCommit(
           () ->
-              updateResult.set(
+              updateResult.addAndGet(
                   SessionUtils.doWithoutCommitAndFetchResult(
                       ModelVersionMetaMapper.class,
                       mapper ->
                           mapper.updateModelVersionMeta(
                               POConverters.updateModelVersionPO(
-                                  oldModelVersionPO, newModelVersionEntity),
-                              oldModelVersionPO))),
+                                  oldModelVersionPOs.get(0), newModelVersionEntity),
+                              oldModelVersionPOs.get(0)))),
+          // TODO Only modifying the unknown URI is supported currently.
+          () ->
+              oldModelVersionPOs.stream()
+                  .filter(
+                      v ->
+                          ModelVersion.URI_NAME_UNKNOWN.equals(v.getModelVersionUriName())
```

Comment:
The condition is quite complicated. Could you please provide a method with a meaningful name to replace it?